### PR TITLE
Allow decoding Event History Download through the UI

### PR DESCRIPTION
**Pull request by Alex-Tideman**
## Description & motivation 💭 
Previously we had an env var to allow downloading decoded event histories. This PR removes that env var and replaces it with a toggle from the UI. Decoding is defaulted on.

This also strips the extra UI attributes we add (id, name, classification, timestamp, category, attributes) from downloaded event histories and JSON view events.

### Screenshots (if applicable) 📸 
Screen Shot 2023-10-23 at 1 52 42 PM

### How was this tested 👻 

- [x] Manual testing
- [ ] E2E tests added
- [x] Unit tests added

### CODE REVIEW

File: src/lib/layouts/workflow-history-layout.svelte [hunk 1]
```diff
@@ -164,3 +174,26 @@
...
+  on:cancelModal={() => (showDownloadPrompt = false)}
+>
+  <h3 slot="title">
+    {translate('common.download-json')}
+  </h3>
+  <div slot="content" class="flex flex-col gap-4">
+    <ToggleSwitch
+      label={translate('events.decode-event-history')}
+      id="decode-event-history"
+      bind:checked={decodeEventHistory}
```
Comment (laurakwhit) [inline src/lib/layouts/workflow-history-layout.svelte:195]: Might be nice to default this to whatever the user has selected for `View decoded event history`

Screenshot 2023-10-23 at 3 07 54 PM

File: src/lib/components/workflow/workflow-json-navigator.svelte [hunk 2]
```diff
@@ -96,7 +100,11 @@
   {:else}
     {#key index}
       <CodeBlock
```
Comment (laurakwhit) [inline src/lib/components/workflow/workflow-json-navigator.svelte:102]: Looks like this is existing behavior, but when toggling `View decoded event history` off and then back on again it looks like there is a page "flash" and a full UI re-render. Might be something we can improve...

Screenshot 2023-10-23 at 3 10 13 PM

**Review by laurakwhit — APPROVED** [review]

File: src/lib/components/workflow/workflow-json-navigator.svelte (hunk 2 above)
Comment (Alex-Tideman) [reply src/lib/components/workflow/workflow-json-navigator.svelte:102]: Good call. We need to rerender the CodeBlock on decode toggle, but I changed the initial decodeValue string from empty string to stringifyWithBigInt(value). So the initial value is the event un-decoded, and then if it decodes successfully, replaces it with the decoded value. This way you don't get that flash which is the change in height from empty string to the decoded value. Much better experience.

Comment (Alex-Tideman) [reply src/lib/components/workflow/workflow-json-navigator.svelte:102]: I also changed the label to "Decode Event History" to make them consistent

File: src/lib/layouts/workflow-history-layout.svelte (hunk 1 above)
Comment (Alex-Tideman) [reply src/lib/layouts/workflow-history-layout.svelte]: 👍 Made it a persistStore and used for both

**Pull request merged**